In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
df = pd.read_csv('amazon.csv')
df.head()

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                                            category discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             ₹399   
1  Computers&Accessories|Accessories&Peripherals|...             ₹199   
2  Computers&Accessories|Accessories&Peripherals|...             ₹199   
3  Computers&Accessories|Accessories&Peripherals|...             ₹329   
4  Computers&Accessories|Accessories&Peripherals|...             ₹154   

  actual_price discount_percentage rating rating_count  \
0       ₹1,099                 64%    4.2       24,269   
1         ₹349                 43%    4.0       43,994   
2       ₹1,899                 90%    3.9        7,928   
3         ₹699                 53%    4.2       94,363   
4         ₹399                 61%    4.2       16,905   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   
3  The boAt Deuce USB 300 2 in 1 cable is compati...   
4  [CHARGE & SYNC FUNCTION]- This cable comes wit...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3  Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4  rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   

                                           review_id  \
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...   
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...   
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...   
3  R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...   
4  R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...   

                                        review_title  \
0  Satisfied,Charging is really fast,Value for mo...   
1  A Good Braided Cable for Your Type C Device,Go...   
2  Good speed for earlier versions,Good Product,W...   
3  Good product,Good one,Nice,Really nice product...   
4  As good as original,Decent,Good one for second...   

                                      review_content  \
0  Looks durable Charging is fine tooNo complains...   
1  I ordered this cable to connect my phone to An...   
2  Not quite durable and sturdy,https://m.media-a...   
3  Good product,long wire,Charges good,Nice,I bou...   
4  Bought this instead of original apple, does th...   

                                            img_link  \
0  https://m.media-amazon.com/images/W/WEBP_40237...   
1  https://m.media-amazon.com/images/W/WEBP_40237...   
2  https://m.media-amazon.com/images/W/WEBP_40237...   
3  https://m.media-amazon.com/images/I/41V5FtEWPk...   
4  https://m.media-amazon.com/images/W/WEBP_40237...   

                                        product_link  
0  https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...  
1  https://www.amazon.in/Ambrane-Unbreakable-Char...  
2  https://www.amazon.in/Sounce-iPhone-Charging-C...  
3  https://www.amazon.in/Deuce-300-Resistant-Tang...  
4  https://www.amazon.in/Portronics-Konnect-POR-1...

#### Understanding data

In [3]:
# Check name of columns
df.columns

Index(['product_id', 'product_name', 'category', 'discounted_price',
       'actual_price', 'discount_percentage', 'rating', 'rating_count',
       'about_product', 'user_id', 'user_name', 'review_id', 'review_title',
       'review_content', 'img_link', 'product_link'],
      dtype='object')

In [4]:
# Check the size of data (rows and columns)
df.shape

(1465, 16)

In [5]:
# Check the data type in columns
df.dtypes

product_id             object
product_name           object
category               object
discounted_price       object
actual_price           object
discount_percentage    object
rating                 object
rating_count           object
about_product          object
user_id                object
user_name              object
review_id              object
review_title           object
review_content         object
img_link               object
product_link           object
dtype: object

In [6]:
# Check the data uniqueness and frequency
df.describe()

product_id                                       product_name  \
count         1465                                               1465   
unique        1351                                               1337   
top     B07JW9H4J1  Fire-Boltt Ninja Call Pro Plus 1.83" Smart Wat...   
freq             3                                                  5   

                                                 category discounted_price  \
count                                                1465             1465   
unique                                                211              550   
top     Computers&Accessories|Accessories&Peripherals|...             ₹199   
freq                                                  233               53   

       actual_price discount_percentage rating rating_count  \
count          1465                1465   1465         1463   
unique          449                  92     28         1143   
top            ₹999                 50%    4.1        9,378   
freq            120                  56    244            9   

                                            about_product  \
count                                                1465   
unique                                               1293   
top     [CHARGE & SYNC FUNCTION]- This cable comes wit...   
freq                                                    6   

                                                  user_id  \
count                                                1465   
unique                                               1194   
top     AHIKJUDTVJ4T6DV6IUGFYZ5LXMPA,AE55KTFVNXYFD5FPY...   
freq                                                   10   

                                                user_name  \
count                                                1465   
unique                                               1194   
top     $@|\|TO$|-|,Sethu madhav,Akash Thakur,Burger P...   
freq                                                   10   

                                                review_id  \
count                                                1465   
unique                                               1194   
top     R3F4T5TRYPTMIG,R3DQIEC603E7AY,R1O4Z15FD40PV5,R...   
freq                                                   10   

                                             review_title  \
count                                                1465   
unique                                               1194   
top     Worked on iPhone 7 and didn’t work on XR,Good ...   
freq                                                   10   

                                           review_content  \
count                                                1465   
unique                                               1212   
top     I am not big on camera usage, personally. I wa...   
freq                                                    8   

                                                 img_link  \
count                                                1465   
unique                                               1412   
top     https://m.media-amazon.com/images/I/413sCRKobN...   
freq                                                    3   

                                             product_link  
count                                                1465  
unique                                               1465  
top     https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...  
freq                                                    1

### Data Wrangling
#### 1. Change the currency unit "₹"
#### 2. Change the data type ("object") to the type we need

In [7]:
df['discounted_price'] = df['discounted_price'].str.replace("₹",'')
df['discounted_price'] = df['discounted_price'].str.replace(",",'')
df['discounted_price'] = df['discounted_price'].astype('float64')

df['actual_price'] = df['actual_price'].str.replace("₹",'')
df['actual_price'] = df['actual_price'].str.replace(",",'')
df['actual_price'] = df['actual_price'].astype('float64')
df.head()

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                                            category  discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             399.0   
1  Computers&Accessories|Accessories&Peripherals|...             199.0   
2  Computers&Accessories|Accessories&Peripherals|...             199.0   
3  Computers&Accessories|Accessories&Peripherals|...             329.0   
4  Computers&Accessories|Accessories&Peripherals|...             154.0   

   actual_price discount_percentage rating rating_count  \
0        1099.0                 64%    4.2       24,269   
1         349.0                 43%    4.0       43,994   
2        1899.0                 90%    3.9        7,928   
3         699.0                 53%    4.2       94,363   
4         399.0                 61%    4.2       16,905   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   
3  The boAt Deuce USB 300 2 in 1 cable is compati...   
4  [CHARGE & SYNC FUNCTION]- This cable comes wit...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3  Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4  rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   

                                           review_id  \
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...   
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...   
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...   
3  R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...   
4  R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...   

                                        review_title  \
0  Satisfied,Charging is really fast,Value for mo...   
1  A Good Braided Cable for Your Type C Device,Go...   
2  Good speed for earlier versions,Good Product,W...   
3  Good product,Good one,Nice,Really nice product...   
4  As good as original,Decent,Good one for second...   

                                      review_content  \
0  Looks durable Charging is fine tooNo complains...   
1  I ordered this cable to connect my phone to An...   
2  Not quite durable and sturdy,https://m.media-a...   
3  Good product,long wire,Charges good,Nice,I bou...   
4  Bought this instead of original apple, does th...   

                                            img_link  \
0  https://m.media-amazon.com/images/W/WEBP_40237...   
1  https://m.media-amazon.com/images/W/WEBP_40237...   
2  https://m.media-amazon.com/images/W/WEBP_40237...   
3  https://m.media-amazon.com/images/I/41V5FtEWPk...   
4  https://m.media-amazon.com/images/W/WEBP_40237...   

                                        product_link  
0  https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...  
1  https://www.amazon.in/Ambrane-Unbreakable-Char...  
2  https://www.amazon.in/Sounce-iPhone-Charging-C...  
3  https://www.amazon.in/Deuce-300-Resistant-Tang...  
4  https://www.amazon.in/Portronics-Konnect-POR-1...

In [8]:
df.isna().sum() # Always use isna() over isnull()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           2
about_product          0
user_id                0
user_name              0
review_id              0
review_title           0
review_content         0
img_link               0
product_link           0
dtype: int64

In [9]:
# Some missing values in rating_count.
nan_cols = [i for i in df.columns if df[i].isnull().any()] # This identify the name of the column
rows_with_nan = df[df.isna().any(axis=1)]
rows_with_nan

product_id                                       product_name  \
282  B0B94JPY2N  Amazon Brand - Solimo 65W Fast Charging Braide...   
324  B0BQRJ3C47  REDTECH USB-C to Lightning Cable 3.3FT, [Apple...   

                                              category  discounted_price  \
282  Computers&Accessories|Accessories&Peripherals|...             199.0   
324  Computers&Accessories|Accessories&Peripherals|...             249.0   

     actual_price discount_percentage rating rating_count  \
282         999.0                 80%    3.0          NaN   
324         999.0                 75%    5.0          NaN   

                                         about_product  \
282  USB C to C Cable: This cable has type C connec...   
324  💎[The Fastest Charge] - This iPhone USB C cabl...   

                          user_id    user_name      review_id  \
282  AE7CFHY23VAJT2FI4NZKKP6GS2UQ       Pranav   RUB7U91HVZ30   
324  AGJC5O5H5BBXWUV7WRIEIOOR3TVQ  Abdul Gafur  RQXD5SAMMPC6L   

                                     review_title  \
282  The cable works but is not 65W as advertised   
324                               Awesome Product   

                                        review_content  \
282  I have a pd supported car charger and I bought...   
324  Quick delivery.Awesome ProductPacking was good...   

                                              img_link  \
282  https://m.media-amazon.com/images/W/WEBP_40237...   
324  https://m.media-amazon.com/images/I/31-q0xhaTA...   

                                          product_link  
282  https://www.amazon.in/Amazon-Brand-Charging-Su...  
324  https://www.amazon.in/REDTECH-Lightning-Certif...

In [10]:
# Duplicate the df for operation. Remove the unwanted info in the operation part
df_op = df.copy().drop(['user_name','review_id','img_link','product_link'], axis=1)
# Replace the NaN with 0
df_op['rating_count'] = df_op['rating_count'].fillna(0)
df_fillna_check = df_op.iloc[282]
df_fillna_check

product_id                                                    B0B94JPY2N
product_name           Amazon Brand - Solimo 65W Fast Charging Braide...
category               Computers&Accessories|Accessories&Peripherals|...
discounted_price                                                   199.0
actual_price                                                       999.0
discount_percentage                                                  80%
rating                                                               3.0
rating_count                                                           0
about_product          USB C to C Cable: This cable has type C connec...
user_id                                     AE7CFHY23VAJT2FI4NZKKP6GS2UQ
review_title                The cable works but is not 65W as advertised
review_content         I have a pd supported car charger and I bought...
Name: 282, dtype: object

In [11]:
# Take care of the category. Need to split them
df_op['category'].value_counts()

category
Computers&Accessories|Accessories&Peripherals|Cables&Accessories|Cables|USBCables                                          233
Electronics|WearableTechnology|SmartWatches                                                                                 76
Electronics|Mobiles&Accessories|Smartphones&BasicMobiles|Smartphones                                                        68
Electronics|HomeTheater,TV&Video|Televisions|SmartTelevisions                                                               63
Electronics|Headphones,Earbuds&Accessories|Headphones|In-Ear                                                                52
                                                                                                                          ... 
Electronics|Cameras&Photography|Accessories|Batteries&Chargers|BatteryChargers                                               1
Computers&Accessories|NetworkingDevices|DataCards&Dongles                                             

In [12]:
df_cat_split = df_op['category'].str.split("|", expand=True).rename(columns={0:'category_1', 1:'category_2', 2:'category_3', 3:'category_4',4:'category_5',5:'category_6', 6:'category_7'})
df_cat_split['category_6'].isna().sum()

1380

###### Total 1465, so we still have 1465-1380 in category_6

In [13]:
df_cat_split['category_7'].isna().sum()

1452

###### Total 1465, so we only have 13 items in category_7

In [14]:
rows_with_value_incat7 = df_cat_split[df_cat_split['category_7'].notna()]
rows_with_value_incat7


category_1              category_2               category_3  \
616   OfficeProducts     OfficePaperProducts                    Paper   
757   OfficeProducts     OfficePaperProducts                    Paper   
765   OfficeProducts     OfficePaperProducts                    Paper   
850   OfficeProducts     OfficePaperProducts                    Paper   
858   OfficeProducts     OfficePaperProducts                    Paper   
878   OfficeProducts     OfficePaperProducts                    Paper   
893   OfficeProducts     OfficePaperProducts                    Paper   
919   OfficeProducts     OfficePaperProducts                    Paper   
955   OfficeProducts     OfficePaperProducts                    Paper   
993   OfficeProducts     OfficePaperProducts                    Paper   
1004  OfficeProducts     OfficePaperProducts                    Paper   
1005  OfficeProducts     OfficePaperProducts                    Paper   
1444    Home&Kitchen  Kitchen&HomeAppliances  Vacuum,Cleaning&Ironing   

             category_4                    category_5    category_6  \
616          Stationery  Pens,Pencils&WritingSupplies  Pens&Refills   
757          Stationery  Pens,Pencils&WritingSupplies  Pens&Refills   
765          Stationery  Pens,Pencils&WritingSupplies  Pens&Refills   
850          Stationery  Pens,Pencils&WritingSupplies  Pens&Refills   
858          Stationery  Pens,Pencils&WritingSupplies  Pens&Refills   
878          Stationery  Pens,Pencils&WritingSupplies  Pens&Refills   
893          Stationery  Pens,Pencils&WritingSupplies  Pens&Refills   
919          Stationery  Pens,Pencils&WritingSupplies  Pens&Refills   
955          Stationery  Pens,Pencils&WritingSupplies  Pens&Refills   
993          Stationery  Pens,Pencils&WritingSupplies  Pens&Refills   
1004         Stationery  Pens,Pencils&WritingSupplies  Pens&Refills   
1005         Stationery  Pens,Pencils&WritingSupplies  Pens&Refills   
1444  Vacuums&FloorCare             VacuumAccessories    VacuumBags   

                    category_7  
616       GelInkRollerballPens  
757                 BottledInk  
765   RetractableBallpointPens  
850         StickBallpointPens  
858   RetractableBallpointPens  
878                 BottledInk  
893       GelInkRollerballPens  
919         StickBallpointPens  
955         StickBallpointPens  
993    LiquidInkRollerballPens  
1004   LiquidInkRollerballPens  
1005              FountainPens  
1444              HandheldBags

###### Interestingly, except the lst one(1444), all other items are from the similar category ('Pens&Refills')
###### According to this result, the category_7 is actually not that important

In [15]:
def replace_cat(category:str):
    df_op[category] = df_cat_split[category]

for i in range(1,7):
    x = "category_" + str(i)
    replace_cat(x)

df_op.tail()

product_id                                       product_name  \
1460  B08L7J3T31  Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...   
1461  B01M6453MB  Prestige Delight PRWO Electric Rice Cooker (1 ...   
1462  B009P2LIL4  Bajaj Majesty RX10 2000 Watts Heat Convector R...   
1463  B00J5DYCCA  Havells Ventil Air DSP 230mm Exhaust Fan (Pist...   
1464  B01486F4G6  Borosil Jumbo 1000-Watt Grill Sandwich Maker (...   

                                               category  discounted_price  \
1460  Home&Kitchen|Kitchen&HomeAppliances|WaterPurif...             379.0   
1461  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...            2280.0   
1462  Home&Kitchen|Heating,Cooling&AirQuality|RoomHe...            2219.0   
1463  Home&Kitchen|Heating,Cooling&AirQuality|Fans|E...            1399.0   
1464  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...            2863.0   

      actual_price discount_percentage rating rating_count  \
1460         919.0                 59%      4        1,090   
1461        3045.0                 25%    4.1        4,118   
1462        3080.0                 28%    3.6          468   
1463        1890.0                 26%      4        8,031   
1464        3690.0                 22%    4.3        6,987   

                                          about_product  \
1460  SUPREME QUALITY 90 GRAM 3 LAYER THIK PP SPUN F...   
1461                       230 Volts, 400 watts, 1 Year   
1462  International design and styling|Two heat sett...   
1463  Fan sweep area: 230 MM ; Noise level: (40 - 45...   
1464  Brand-Borosil, Specification – 23V ~ 5Hz;1 W...   

                                                user_id  \
1460  AHITFY6AHALOFOHOZEOC6XBP4FEA,AFRABBODZJZQB6Z4U...   
1461  AFG5FM3NEMOL6BNFRV2NK5FNJCHQ,AGEINTRN6Z563RMLH...   
1462  AGVPWCMAHYQWJOQKMUJN4DW3KM5Q,AF4Q3E66MY4SR7YQZ...   
1463  AF2JQCLSCY3QJATWUNNHUSVUPNQQ,AFDMLUXC5LS5RXDJS...   
1464  AFGW5PT3R6ZAVQR4Y5MWVAKBZAYA,AG7QNJ2SCS5VS5VYY...   

                                           review_title  \
1460  Received the product without spanner,Excellent...   
1461  ok,everything was good couldn't return bcoz I ...   
1462  very good,Work but front melt after 2 month,Go...   
1463  Fan Speed is slow,Good quality,Good product,go...   
1464  Works perfect,Ok good product,Nice Product. Re...   

                                         review_content    category_1  \
1460  I received product without spanner,Excellent p...  Home&Kitchen   
1461  ok,got everything as mentioned but the measuri...  Home&Kitchen   
1462  plastic but cool body ,u have to find sturdy s...  Home&Kitchen   
1463  I have installed this in my kitchen working fi...  Home&Kitchen   
1464  It does it job perfectly..only issue is temp c...  Home&Kitchen   

                      category_2                  category_3  \
1460      Kitchen&HomeAppliances  WaterPurifiers&Accessories   
1461      Kitchen&HomeAppliances      SmallKitchenAppliances   
1462  Heating,Cooling&AirQuality                 RoomHeaters   
1463  Heating,Cooling&AirQuality                        Fans   
1464      Kitchen&HomeAppliances      SmallKitchenAppliances   

                    category_4 category_5 category_6  
1460  WaterPurifierAccessories       None       None  
1461         Rice&PastaCookers       None       None  
1462            HeatConvectors       None       None  
1463               ExhaustFans       None       None  
1464            SandwichMakers       None       None

In [20]:
df_op.drop(columns = 'category')

product_id                                       product_name  \
0     B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1     B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2     B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3     B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4     B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   
...          ...                                                ...   
1460  B08L7J3T31  Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...   
1461  B01M6453MB  Prestige Delight PRWO Electric Rice Cooker (1 ...   
1462  B009P2LIL4  Bajaj Majesty RX10 2000 Watts Heat Convector R...   
1463  B00J5DYCCA  Havells Ventil Air DSP 230mm Exhaust Fan (Pist...   
1464  B01486F4G6  Borosil Jumbo 1000-Watt Grill Sandwich Maker (...   

      discounted_price  actual_price discount_percentage rating rating_count  \
0                399.0        1099.0                 64%    4.2       24,269   
1                199.0         349.0                 43%    4.0       43,994   
2                199.0        1899.0                 90%    3.9        7,928   
3                329.0         699.0                 53%    4.2       94,363   
4                154.0         399.0                 61%    4.2       16,905   
...                ...           ...                 ...    ...          ...   
1460             379.0         919.0                 59%      4        1,090   
1461            2280.0        3045.0                 25%    4.1        4,118   
1462            2219.0        3080.0                 28%    3.6          468   
1463            1399.0        1890.0                 26%      4        8,031   
1464            2863.0        3690.0                 22%    4.3        6,987   

                                          about_product  \
0     High Compatibility : Compatible With iPhone 12...   
1     Compatible with all Type C enabled devices, be...   
2     【 Fast Charger& Data Sync】-With built-in safet...   
3     The boAt Deuce USB 300 2 in 1 cable is compati...   
4     [CHARGE & SYNC FUNCTION]- This cable comes wit...   
...                                                 ...   
1460  SUPREME QUALITY 90 GRAM 3 LAYER THIK PP SPUN F...   
1461                       230 Volts, 400 watts, 1 Year   
1462  International design and styling|Two heat sett...   
1463  Fan sweep area: 230 MM ; Noise level: (40 - 45...   
1464  Brand-Borosil, Specification – 23V ~ 5Hz;1 W...   

                                                user_id  \
0     AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1     AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2     AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3     AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4     AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   
...                                                 ...   
1460  AHITFY6AHALOFOHOZEOC6XBP4FEA,AFRABBODZJZQB6Z4U...   
1461  AFG5FM3NEMOL6BNFRV2NK5FNJCHQ,AGEINTRN6Z563RMLH...   
1462  AGVPWCMAHYQWJOQKMUJN4DW3KM5Q,AF4Q3E66MY4SR7YQZ...   
1463  AF2JQCLSCY3QJATWUNNHUSVUPNQQ,AFDMLUXC5LS5RXDJS...   
1464  AFGW5PT3R6ZAVQR4Y5MWVAKBZAYA,AG7QNJ2SCS5VS5VYY...   

                                           review_title  \
0     Satisfied,Charging is really fast,Value for mo...   
1     A Good Braided Cable for Your Type C Device,Go...   
2     Good speed for earlier versions,Good Product,W...   
3     Good product,Good one,Nice,Really nice product...   
4     As good as original,Decent,Good one for second...   
...                                                 ...   
1460  Received the product without spanner,Excellent...   
1461  ok,everything was good couldn't return bcoz I ...   
1462  very good,Work but front melt after 2 month,Go...   
1463  Fan Speed is slow,Good quality,Good product,go...   
1464  Works perfect,Ok good product,Nice Product. Re...   

                                         review_content  \
0     

In [21]:
df_op['category_1'].value_counts()

category_1
Electronics              526
Computers&Accessories    453
Home&Kitchen             448
OfficeProducts            31
MusicalInstruments         2
HomeImprovement            2
Toys&Games                 1
Car&Motorbike              1
Health&PersonalCare        1
Name: count, dtype: int64

In [30]:
# Noticed that the user_id column also needs to be split
df_user_split = df_op['user_id'].str.split(',', expand=True).rename(columns={0:'user_1', 1:'user_2', 2:'user_3', 3:'user_4',4:'user_5',5:'user_6', 6:'user_7', 7:'user_8'})
df_user_split.head()

# We can edit the df_op, but I don't thiink we need it for now.
# def replace_user(user_id:str):
#     df_op[user_id] = df_user_split[user_id]

# for i in range(1,9):
#     x = "user_" + str(i)
#     replace_user(x)

# df_op.tail()

user_1                        user_2  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q  AHMY5CWJMMK5BJRBBSNLYT3ONILA   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ  AGYYVPDD7YG7FYNBXNGXZJT525AQ   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA  AESFLDV2PT363T2AQLWQOWZ4N3OA   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A  AG5HTSFRRE6NL3M5SGCUQBP7YSCA   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA  AFUGIFH5ZAFXRDSZHM4QB2KPKFUQ   

                         user_3                        user_4  \
0  AHCTC6ULH4XB6YHDY6PCH2R772LQ  AGYHHIERNXKA6P5T7CZLXKVPT7IQ   
1  AHONIZU3ICIEHQIGQ6R2VFRSBXOQ  AFPHD2CRPDZMWMBL7WXRSVYWS5JA   
2  AHTPQRIMGUD4BYR5YIHBH3CCGEFQ  AEUVWXYP5LT7PZLLZENEO2NODPBQ   
3  AH725ST5NW2Y4JZPKUNTIJCUK2BA  AHV3TXIFCJPMS4D5JATCEUR266MQ   
4  AFK4NJOLFSJGWLOJIUIAROJF6YVA  AFUOTYRFUXVPEBGIXVZZ7DR3CZUA   

                         user_5                        user_6  \
0  AG4OGOFWXJZTQ2HKYIOCOY3KXF2Q  AENGU523SXMOS7JPDTW52PNNVWGQ   
1  AEZ346GX3HJ4O4XNRPHCNHXQURMQ  AEPSWFPNECKO34PUC7I56ITGXR6Q   
2  AHC7MPW55DOO6WNCOQVA2VHOD26A  AFDI6FRPFBTNBG7BAEB7JDJSMKDQ   
3  AGWIGDEMFIIUAOXYY2QATNBSUGHA  AFSTSLQUV4EVEXWKBOLEFHL2H5YQ   
4  AFDLRSXKDZ6U3U3KD46SQLFGZQRA  AH5VLM66SIK7J3IRG4NY7XVOQ55A   

                         user_7                        user_8  
0  AEQJHCVTNINBS4FKTBGQRQTGTE5Q  AFC3FFC5PKFF5PMA52S3VCHOZ5FQ  
1  AHWVEHR5DYLVFTO2KF3IZATFQSWQ  AH4QT33M55677I7ISQOAKEQWACYQ  
2  AFQKCEEEKXCOHTDG4WUN3XPPHJQQ  AHKUUFNMBZIDLSSPA4FEHIO2EC7Q  
3  AGAKDNBHY2FKX7I4ACRGILU7QL7A  AFNWJUWJRHCC6HN52KMG5AKZY37Q  
4  AE3MQNNHHLUHXURL5S7IAR7JTGNQ  AFSEOFZY67MYC7UAJU264Z5NFTLA